### PR TITLE
Task/bws30 groups management

### PR DESCRIPTION
**Pull request by Donymak**

### CODE REVIEW

File: backend/src/common/types/group/group.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export { type GroupDto } from 'bws-shared/common/types/types';
```
Comment (Yaroslaww-1) [inline backend/src/common/types/group/group.ts:1]: GroupResponseDto

File: backend/src/services/group/group.entity.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    this.name = name;
+    this.createdAt = createdAt;
+    this.tenant_id = tenant_id;
+  }
+
+  public static initialize({
+    id,
+    name,
+    createdAt,
+    tenant_id,
```
Comment (Yaroslaww-1) [inline backend/src/services/group/group.entity.ts:30]: tenantId

**Review by Yaroslaww-1 — CHANGES_REQUESTED** [review]

File: backend/src/data/repositories/group/group.repository.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  constructor({ GroupModel }: Constructor) {
+    this.#GroupModel = GroupModel;
+  }
+
+  async getAll(): Promise<GroupEntity[]> {
+    const groups = await this.#GroupModel.query();
+    return groups.map(Group.modelToEntity);
+  }
+
+  async getByName(name: string): Promise<GroupEntity | null> {
```
Comment (what1s1ove) [inline backend/src/data/repositories/group/group.repository.ts:20]: not sure that you use it

File: backend/src/services/group/group.entity.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  }: {
+    id: string;
+    name: string;
+    createdAt: Date;
+    tenant_id: string;
+  }) {
+    this.id = id;
+    this.name = name;
+    this.createdAt = createdAt;
+    this.tenant_id = tenant_id;
```
Comment (what1s1ove) [inline backend/src/services/group/group.entity.ts:23]:
```suggestion
    this.tenant_id = tenantId;
```

File: backend/src/services/group/group.entity.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    tenant_id,
+  }: {
+    name: string;
+    tenant_id: string;
+  }): Group {
+    return new Group({
+      id: getRandomId(),
+      name,
+      createdAt: new Date(),
+      tenant_id,
```
Comment (what1s1ove) [inline backend/src/services/group/group.entity.ts:56]:
```suggestion
      tenantId,
```

File: backend/src/services/group/group.service.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+};
+
+class Group {
+  #groupRepository: typeof groupRep;
+
+  constructor({ groupRepository }: Constructor) {
+    this.#groupRepository = groupRepository;
+  }
+
+  async getAll(): Promise<GroupDto[]> {
```
Comment (what1s1ove) [inline backend/src/services/group/group.service.ts:16]: not sure that we use it

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/data/repositories/repositories.ts [hunk 7]
```diff
@@ -1,8 +1,16 @@
...
+} from '~/data/models/models';
 import { Master } from './master/master.repository';
+import { Group } from './group/group.repository';
 
 const master = new Master({
   MasterModel,
 });
 
-export { master };
+const group = new Group({
```
Comment (Yaroslaww-1) [inline backend/src/data/repositories/repositories.ts:12]: groupRepository

File: backend/src/data/repositories/group/group.repository.ts [hunk 8]
```diff
@@ -0,0 +1,52 @@
+import { Group as GroupM } from '~/data/models/models';
+import { Group as GroupEntity } from '~/services/group/group.entity';
+
+type Constructor = {
+  GroupModel: typeof GroupM;
+};
+
+class Group {
```
Comment (Yaroslaww-1) [inline backend/src/data/repositories/group/group.repository.ts:8]: GroupEntity

**Review by Yaroslaww-1 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**